man/fi_poll.3.md

```diff
@@ -159,9 +170,10 @@ struct fi_wait_attr {
 
 - *FI_WAIT_FD*
 : Indicates that the wait set should use a file descriptor as its wait
-  mechanism.  A file descriptor wait object must be usable in select,
-  poll, and epoll routines.  However, a provider may signal an FD wait
-  object by marking it as readable, writable, or with an error.
+  mechanism. Since it may not always be possible to use a single underlying
+  wait object, the file descriptor is not guaranteed to be usable in select,
+  poll, and epoll routines. However, when possible, a provider may signal an FD
+  wait object by marking it as readable, writable, or with an error.
```

Comment:
> I wonder if we can declare that at least all wait objects must be fd's, and that fi_control can be used to return all of them?

So if the provider is internally using epoll it would return a list of the FDs associated with the epoll object rather than the epoll FD itself?